ddtrace/internal/packages.py

```diff
@@ -190,17 +190,22 @@ def is_namespace(f: metadata.PackagePath):
     try:
         mapping = {}
 
-        for dist in metadata.distributions():
-            if dist is not None and dist.files is not None:
-                d = Distribution(name=dist.metadata["name"], version=dist.version, path=None)
-                for f in dist.files:
-                    root = f.parts[0]
-                    if root.endswith(".dist-info") or root.endswith(".egg-info") or root == "..":
-                        continue
-                    if is_namespace(f):
-                        root = "/".join(f.parts[:2])
-                    if root not in mapping:
-                        mapping[root] = d
+        for dist in importlib_metadata.distributions():
+            files = dist.files
+            if not files:
+                continue
+            metadata = dist.metadata
+            name = metadata["name"]
+            version = metadata["version"]
```

Comment:
Can we save these two assignment operations perhaps?

Reply:
Done